frame/esg/src/lib.rs

```diff
@@ -179,23 +180,14 @@ pub mod pallet {
 				.clamp(0, MAX_ESG_SCORE)
 		}
 
-		/// tries to parse accoundId from given json value under key 'account'
-		/// returns Some(AccountId) if everything is ok or
-		/// returns None if something, like invalid string provided, about it is not ok
-		pub fn try_parse_acc_id(acc_val: Option<&serde_json::Value>) -> Option<T::AccountId> {
+		pub fn try_parse_addr(acc_val: Option<&serde_json::Value>) -> Option<AccIdEth> {
 			let acc = acc_val.unwrap_or(&Value::Null).as_str().unwrap_or("");
 
-			if Self::not_valid_acc_id_format(acc.as_bytes()) {
+			if Self::not_valid_addr(acc.as_bytes()) {
 				return None
 			}
-
-			// str to <T as Config>::AccountId
-			let mut decoded = [0; 48];
-			bs58::decode(acc).into(&mut decoded).unwrap_or(0);
-			match T::AccountId::decode(&mut &decoded[1..33]) {
-				Ok(v) => Some(v),
-				_ => None,
-			}
+			let acc = Self::hexstr2acc_id20(&acc[2..]);
```

Comment:
we just decode correctly format . I mean convert string `hex` evm address to `AccountId` type (take 20 bytes)

Reply:
internally T::AccontId is of AccountId32, so that is y i am using AccountId20::from(). You must check the evm pallet as a reference

Reply:
I don't think so, our runtime is using `AccountId20`